dask/dataframe/rolling.py

```diff
@@ -91,31 +146,174 @@ def map_overlap(func, df, before, after, *args, **kwargs):
         ):
             raise ValueError("before and after must be positive integers")
 
-    if "token" in kwargs:
-        func_name = kwargs.pop("token")
-        token = tokenize(df, before, after, *args, **kwargs)
+    name = kwargs.pop("token", None)
+    parent_meta = kwargs.pop("parent_meta", None)
+
+    assert callable(func)
+    if name is not None:
+        token = tokenize(meta, *args, **kwargs)
     else:
-        func_name = "overlap-" + funcname(func)
-        token = tokenize(func, df, before, after, *args, **kwargs)
+        name = funcname(func)
+        token = tokenize(func, meta, *args, **kwargs)
+    name = f"{name}-{token}"
+
+    from dask.array.core import normalize_arg
+    from dask.dataframe.core import (
+        Scalar,
+        _get_divisions,
+        _get_meta,
+        _is_only_scalar,
+        _maybe_from_pandas,
+        apply_and_enforce,
+        new_dd_object,
+        partitionwise_graph,
+    )
+    from dask.dataframe.multi import _maybe_align_partitions
+    from dask.delayed import unpack_collections
+    from dask.utils import apply
+
+    if align_dataframes:
+        args = _maybe_from_pandas(args)
+        args = _maybe_align_partitions(args)
```

Comment:
Minor comment, #9075 added a helpful error message to set `align_dataframes=False`  for `map_partitions`, we would want something similar here as well?

Reply:
We think it's alight to duplicate code for this bit, especially because we're trying to reduce the number of super-specific helper functions